### PR TITLE
Test birth demographic cache

### PR DESCRIPTION
**Pull request by labkey-martyp**
#### Rationale
Add a test covering assignment inheritance from dam to infant at birth and verify snapshot is accurately updated. Add some sample data similar to base EHR. Fix some flaky tests.

#### Related Pull Requests
* https://github.com/LabKey/ehrModules/pull/781

#### Changes
* Add testBirthInheritFromDam
* Add sample data for easier testing
* Remove Ext3 function from snapshot panel
* Snapshot groups link only if valid group assignment
* Small changes for flaky tests

### CODE REVIEW

File: onprc_ehr/resources/data/divider_types.tsv [hunk 1]
```diff
@@ -0,0 +1,10 @@
+divider	countAsSeparate	countAsPaired	countAsBoth	isMoveable	displaychar	short_description	bgcolor	border_style
+Mesh	TRUE	FALSE		TRUE	M	MESH	grey	dotted
+Solid	TRUE	FALSE		TRUE	S			solid
+Clear	TRUE	FALSE		TRUE	C	CLEAR		solid
+No Slide	TRUE	TRUE		TRUE	NO		yellow	none
+Pairing Divider	TRUE	FALSE		TRUE	P		grey	solid
+								
+								
+								
+
```
Comment (labkey-jeckels) [inline onprc_ehr/resources/data/divider_types.tsv]: Harmless but useless newlines

File: onprc_ehr/resources/web/onprc_ehr/panel/SnapshotPanel.js [hunk 2]
```diff
@@ -284,9 +284,8 @@ Ext4.define('onprc_ehr.panel.SnapshotPanel', {
 
 
         //Modified: 2-14-2017  R.Blasa  Make animal group name as hyperlink
-        var text = values.length ? values.join('<br>') : 'None';
-        var url =  '<a href="' + LABKEY.ActionURL.buildURL('query', 'executeQuery', null, {schemaName: 'ehr', 'query.queryName': 'animal_groups',
-                  'query.name~eq': values.length ? values.join(';') : 'None' }) + '" target="_blank">' + text  + '</a>';
+        var url =  values.length ? '<a href="' + LABKEY.ActionURL.buildURL('query', 'executeQuery', null, {schemaName: 'ehr', 'query.queryName': 'animal_groups',
```
Comment (labkey-jeckels) [inline onprc_ehr/resources/web/onprc_ehr/panel/SnapshotPanel.js:287]: Missing URL and HTML encoding.

**Review by labkey-jeckels — APPROVED** [review]

File: onprc_ehr/resources/data/divider_types.tsv (hunk 1 above)
Comment (labkey-martyp) [reply onprc_ehr/resources/data/divider_types.tsv]: removed

File: onprc_ehr/resources/web/onprc_ehr/panel/SnapshotPanel.js (hunk 2 above)
Comment (labkey-martyp) [reply onprc_ehr/resources/web/onprc_ehr/panel/SnapshotPanel.js:287]: Good catch. I added the HTML encoding. ActionURL.buildURL should automatically URL encode the parameters if that is what your concern is there.

**Review by labkey-sweta — APPROVED** [review]
Approving the test code.

**Pull request merged**